neffint/fixed_grid_fourier_integral.py

```diff
@@ -0,0 +1,179 @@
+    xx = x[taylor_mask]
+    inv_factorial = 1.
+    jx_to_n = np.ones_like(1j*xx) # Starts as x**0
+    abs_x_to_n_plus_1 = np.abs(xx) # Starts as x**(0+1)
+    exp_abs_x = np.exp(np.abs(xx))
+
+    # Run convergence loop until phi and psi converged
+    phi_converged = False
+    psi_converged = False
+    for n in range(MAX_PHI_PSI_ITERATIONS+1):
+
+        # Stop loop if converged
+        if phi_converged and psi_converged:
+            break
+
+        # Only calculate phi if not yet converged (to save computation time)
+        if not phi_converged:
+            phi[taylor_mask] += (n+6)/((n+3)*(n+4)) * jx_to_n * inv_factorial
+
+            # Check for convergence
+            min_phi_component = np.min([np.abs(np.real(phi[taylor_mask])), np.abs(np.imag(phi[taylor_mask]))], axis=0)
+            phi_remaining_terms_bound = 2 * abs_x_to_n_plus_1 * exp_abs_x * inv_factorial / ((n+1)*(n+5))
+            phi_converged = np.all(phi_remaining_terms_bound < rel_tolerance * min_phi_component)
+        
+        # Only calculate pis if not yet converged
+        if not psi_converged:
+            psi[taylor_mask] += - 1/((n+3)*(n+4)) * jx_to_n * inv_factorial
+
+            # Check for convergence
+            min_psi_component = np.min([np.abs(np.real(psi[taylor_mask])), np.abs(np.imag(psi[taylor_mask]))], axis=0)
+            psi_remaining_terms_bound = abs_x_to_n_plus_1 * exp_abs_x * inv_factorial / ((n+1)*(n+4)*(n+5))
+            psi_converged = np.all(psi_remaining_terms_bound < rel_tolerance*min_psi_component)
+        
+        # Update variables
+        jx_to_n *= 1j*xx
+        abs_x_to_n_plus_1 *= np.abs(xx)
+        inv_factorial /= n+1
+
+    # If loop finished without converging, raise error
+    if not (phi_converged and psi_converged):
+        raise RuntimeError(f"Phi and Psi calculation failed to converge after {MAX_PHI_PSI_ITERATIONS} iterations")
+    
+    return phi, psi
+
+def fourier_integral_fixed_sampling_pchip(
+    times: ArrayLike,
+    frequencies: ArrayLike,
+    func_values: ArrayLike,
+    inf_correction_term: bool
+) -> np.ndarray:
+    """Calculates the fourier integral of the function, given as an array of values corresponding to an array of frequencies, for the time values given as input.
+    A Filon type algorithm using a piecewise cubic Hermite interpolating polynomial (pchip) and optionally an asymptotic correction term.
+    For details on implementation, see [1].
+
+    :param times: Float or 1D array of floats of length M, the time(s) to compute the fourier integral for.
+    :type times: ArrayLike
+    :param frequencies: Float or 1D array of floats of length N, the frequencies the function to be transformed have been evaluated at.
+    :type frequencies: ArrayLike
+    :param func_values: Complex or ND array of complex of shape (N, X1, X2, ...), the outputs of the function to be transformed at the frequencies given as input.
+    :type func_values: ArrayLike
```

Comment:
I did not mention units in the docstring (yet). I think it is probably a good idea to mention it, but not sure if I should just write [s] and [Hz] or if I should write that any coherent time and frequency units work (e.g. ns and GHz should work, so should m and m^-1)